erc1155/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+            .or_default()
+            .insert(*account, amount);
+
+        debug!("after mint: {:?}", self.get_balance(account, id));
+    }
+
+    fn balance_of(&self, account: &ActorId, id: &u128) -> u128 {
+        self.get_balance(account, id)
+    }
+
+    fn balance_of_batch(&self, accounts: &[ActorId], ids: &[u128]) {
+        // TODO
+        // return multiple values
+        if accounts.len() != ids.len() {
+            panic!("ERC1155: accounts and ids length mismatch");
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let account = accounts[i];
+            self.get_balance(&account, &ele);
+        }
+    }
+
+    fn mint(&mut self, from: &ActorId, id: &u128, amount: u128) {
+        // check owner
+        if from == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address");
+        }
+        let old_balance = self.get_balance(from, id);
+
+        // self.balances.insert(key: K, value: V)
+
+        self.set_balance(from, id, old_balance.saturating_add(amount));
+
+        // .insert(msg::source(), balance.saturating_add(U256::one()));
+
+        // TransferSingle event
+    }
+
+    fn mint_batch(&mut self, from: &ActorId, ids: &[u128], amounts: &[u128]) {
+        if from == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address");
+        }
+
+        if ids.len() != amounts.len() {
+            panic!("ERC1155: ids and amounts length mismatch");
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let amount = amounts[i];
+            let old_balance = self.get_balance(from, ele);
+            self.set_balance(from, ele, old_balance.saturating_add(amount));
+        }
+
+        // TransferBatch event
+    }
+
+    fn burn_batch(&mut self, from: &ActorId, ids: &[u128], amounts: &[u128]) {
+        // TODO
+        // check owner
+        if from == &ZERO_ID {
+            panic!("ERC1155: burn from the zero address");
+        }
+
+        if ids.len() != amounts.len() {
+            panic!("ERC1155: ids and amounts length mismatch")
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let amount = amounts[i];
+
+            let from_balance = self.get_balance(from, ele);
+
+            if from_balance < amount {
+                panic!("ERC1155: burn amount exceeds balance")
+            }
+
+            self.set_balance(from, ele, from_balance.saturating_sub(amount));
+        }
+
+        // TransferBatch event
+    }
+
+    fn set_approval_for_all(&mut self, owner: &ActorId, operator: &ActorId, approved: bool) {
+        if owner != operator {
+            panic!("ERC1155: setting approval status for self")
+        }
+        self.operator_approvals
+            .entry(*owner)
+            .or_default()
+            .insert(*operator, approved);
+
+        // ApprovalForAll event
+    }
+
+    fn is_approved_for_all(&mut self, account: &ActorId, operator: &ActorId) -> &bool {
+        self.get_approval(account, operator)
```

Comment:
What is this function for if there is already a function `get_approval`?

Reply:
I rename get_approval to _get_approval,  treat it as an internal function.
the `is_approved_for_all` is the public function.